fragalysis_api/xcimporter/align.py

```diff
@@ -98,138 +111,407 @@ def __get_length_and_resolution(self, file):
+                print(i)
+                s2 = time.time()
+                map = Xmap.from_file(file=Path(os.path.join(dir, f'{base}{ext}')))
+                map.resample(xmap=map, transform=transform)
+                map.save(path=Path(os.path.join(out_dir, f'{base}{ext}')))
+                e2 = time.time()
+                print(f'{int(e2 - s2)} seconds to transform map...')
+                e = time.time()
+                print(f'Total Running time: {int(e - s) / 60} minutes.')
+
+    def write_align_ref(self, output):
+        '''
+        Copy the reference pdb structure used for alignment to a new location
+        :param output: The corresponding filename to write the reference pdb to
+        :return: the pdbfile that was chosen as reference copied to the located specified by output
+        '''
+        fn = os.path.join(self.directory, f'{self._get_ref}.pdb')
+        shutil.copyfile(fn, output)
 
     def align(self, out_dir):
         """
-        Aligns all pdbs in the pymol object to the pdb_ref.
+        Aligns all pdb structures and map files (if any) using gemmi, and save them to a new directory.
         :param out_dir: directory to save aligned pdbs in
-        :return: saves the pdbs
+        :return: saves the pdbs + transforms map files (if any!)
         """
-
-        # Silently open PyMOL
-        pymol.pymol_argv = ['pymol', '-qc']
-        pymol.finish_launching()
-        pymol_cmd = self._load_objs()
-
-        for num, name in enumerate(pymol_cmd.get_names()):  # Saves the aligned pdb files from the cmd as pdb files
-
-            if not name == self._get_ref:
-                pymol_cmd.align(name, self._get_ref)
-                self._save_align(name, pymol_cmd.get_pdbstr(selection=name), out_dir)
-
-            elif name == self._get_ref:
-                self._save_align(name, pymol_cmd.get_pdbstr(selection=name), out_dir)
+        # load ref
+        input_files = self._get_files
+        map_list = self._get_maplist
+        crystals = [os.path.splitext(os.path.basename(f))[0] for f in input_files]
+        ref = self._get_ref
+        dir = self.directory
+        mono = self.mono
+
+        s = time.time()
+        for num, name in enumerate(crystals):
+            reference_pdb = Structure.from_file(file=Path(os.path.join(dir, f'{ref}.pdb')))
+            all_maps = [j for j in map_list if name in j]
+            if not name == ref:
+                # Do an alignment + save
+                current_pdb = Structure.from_file(file=Path(os.path.join(dir, f'{name}.pdb')))
+                try:
+                    current_pdb, transform = current_pdb.align_to(other=reference_pdb, monomerized=mono)
+                except Exception as e:
+                    # Poorly Documented. Use better stuff...
+                    print(f'{e}')
+                    continue
+
+                current_pdb.structure.write_pdb(os.path.join(out_dir, f'{name}_bound.pdb'))
+                # Align Xmaps + save!
```

Comment:
Is this repeated from ``align_to_reference``? if it is - separate out into a new function that can be called in both places. Do the same for anywhere else you've copied and pasted code